### PR TITLE
Add a dev dockerfile

### PR DESCRIPTION
**Pull request by jcoyne**
The main dockerfile should be for `RAILS_ENV=production` and thus should be bundled `--without development test`

### CODE REVIEW

File: docker/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# This image's actual base image
+FROM starefossen/ruby-node:2-10
+
+LABEL maintainer="Aaron Collier <aaron.collier@stanford.edu>"
+
+# Set default RAILS environment
+ENV RAILS_ENV=production
```
Comment (justinlittman) [inline docker/Dockerfile.dev]: Why is this production?

**Comment by justinlittman** [comment]
Thinking about this more. Would need to do the same for worker as well. Is is worth maintaining multiple sets of duplicate Dockerfiles just to avoid dev and test deps?

**Comment by jcoyne** [comment]
@justinlittman I've had problems before running `RAILS_ENV=production` when you've got dev and test deps installed. See https://blog.kurttomlinson.com/posts/don-t-install-the-spring-gem-on-your-production-rails-server

**Comment by justinlittman** [comment]
Can we handle the difference with a build arg? Trying to avoid repeating / confusing ourselves.

**Comment by jcoyne** [comment]
@justinlittman I don't know of any way, but maybe you do?

**Comment by justinlittman** [comment]
No, but let me give it a crack.

**Pull request closed without merging**